website_event_membership_restriction/views/event_templates.xml

```diff
@@ -43,10 +45,6 @@
             <attribute name="class">d-none</attribute>
         </xpath>
         <xpath expr="//form//div[@t-else='']//button" position="after">
-            <t
-                t-set="is_disabled_for_non_members"
-                t-value="event.registration_membership_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'] or request.env.user.partner_id.membership_state not in ['paid', 'free'])"
-            />
```

Comment:
If you remove the condition for the second button, the events using this button remain active because the condition is not defined for this button.
NOT member
![image](https://github.com/OCA/event/assets/118818446/f7521589-55e0-4f71-9a1b-1bca586cbbca)
Only members for this event:
![image](https://github.com/OCA/event/assets/118818446/90797011-1d6c-4be2-9b3a-ccc0635ea476)
Can be registered
![image](https://github.com/OCA/event/assets/118818446/acfeb6b0-6ec7-4fe6-83b1-3f7e28869bc4)

Reply:
I'm not removing the condition (as the button still has the `t-att-class`). I'm just removing to evaluate again the same variables that have been already evaluated before.

Reply:
Moved to the beginning to be evaluated for all the cases.